src/overlays/actors/ovl_En_Po_Field/z_en_po_field.c

```diff
@@ -21,85 +44,936 @@ const ActorInit En_Po_Field_InitVars = {
+        this->actor.scale.x = temp_f0;
+    }
+    if (this->actionTimer == 18) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_PO_DEAD2);
+    }
+}
+
+void EnPoField_Disappear(EnPoField* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (this->actionTimer != 0) {
+        this->actionTimer--;
+    }
+    this->actor.shape.rot.y += 0x1000;
+    this->lightColor.a = this->actionTimer * 15.9375f;
+    this->actor.shape.unk_14 = this->lightColor.a;
+    if (this->actionTimer == 0) {
+        EnPoField_SetupWaitForSpawn(this, globalCtx);
+    }
+}
+
+void EnPoField_SoulIdle(EnPoField* this, GlobalContext* globalCtx) {
+    if (this->actionTimer != 0) {
+        this->actionTimer--;
+    }
+    if (this->actor.bgCheckFlags & 1) {
+        EffectSsHahen_SpawnBurst(globalCtx, &this->actor.posRot.pos, 6.0f, 0, 1, 1, 15, 109, 10, D_06004BA0);
+        func_80AD42B0(this);
+    } else if (this->actionTimer == 0) {
+        EnPoField_SetupWaitForSpawn(this, globalCtx);
+    }
+    Actor_MoveForward(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 10.0f, 10.0f, 10.0f, 4);
+}
+
+void EnPoField_SoulUpdateProperties(EnPoField* this, s32 arg1) {
+    EnPoFieldInfo* info = &sPoFieldInfo[this->actor.params];
+    f32 multiplier;
+
+    this->lightColor.a = CLAMP(this->lightColor.a + arg1, 0, 255);
+    if (arg1 < 0) {
+        multiplier = this->lightColor.a * 0.003921569f;
+        this->actor.scale.z = 0.0056000003f * multiplier + 0.00140000007f;
+        this->actor.scale.x = 0.0056000003f * multiplier + 0.00140000007f;
+        this->actor.scale.y = 0.007f - 0.007f * multiplier + 0.007f;
+    } else {
+        multiplier = 1.0f;
+        this->actor.scale.z = this->lightColor.a * 2.7450982e-05f;
+        this->actor.scale.y = this->lightColor.a * 2.7450982e-05f;
+        this->actor.scale.x = this->lightColor.a * 2.7450982e-05f;
+        this->actor.posRot.pos.y = this->actor.initPosRot.pos.y + (0.05882353f * this->lightColor.a);
+    }
+    this->lightColor.r = info->lightColor.r * multiplier;
+    this->lightColor.g = info->lightColor.g * multiplier;
+    this->lightColor.b = info->lightColor.b * multiplier;
+    Lights_PointNoGlowSetInfo(&this->lightInfo, this->actor.posRot.pos.x, this->actor.posRot.pos.y,
+                              this->actor.posRot.pos.z, info->lightColor.r, info->lightColor.g, info->lightColor.b,
+                              this->lightColor.a * 0.78431373f);
+}
+
+void func_80AD587C(EnPoField* this, GlobalContext* globalCtx) {
+    this->actor.initPosRot.pos.y += 2.0f;
+    EnPoField_SoulUpdateProperties(this, 20);
+    if (this->lightColor.a == 255) {
+        func_80AD4384(this);
+    }
+}
+
+void func_80AD58D4(EnPoField* this, GlobalContext* globalCtx) {
+    if (this->actionTimer != 0) {
+        this->actionTimer--;
+    }
+    if (func_8002F194(&this->actor, globalCtx) != 0) {
+        EnPoField_SetupInteractWithSoul(this);
```

Comment:
Not sure what you mean, the only way you know it's from the poe soul's perspective is because of the name to begin with?

Reply:
I guess im confused about what this function/actor is in general. is this not the code for the poe soul itself? in which case it would be interacting with link